lib/node_modules/@stdlib/math/base/special/log10/src/main.c

```diff
@@ -160,56 +69,81 @@ static double klog( const double x ) {
+	if ( hx < HIGH_MIN_NORMAL_EXP ) {
+		// Case: x < 2**-1022
+		if ( ( ( hx & STDLIB_CONSTANT_FLOAT64_HIGH_WORD_ABS_MASK ) | lx ) == 0 ) {
 			return STDLIB_CONSTANT_FLOAT64_NINF;
 		}
-		if ( ihx < 0 ) {
-			return 0.0 / 0.0; // NaN
-		}
-		// Subnormal number, scale up `x`...
 		k -= 54;
+
+		// Subnormal number, scale up x:
 		xc *= TWO54;
 		stdlib_base_float64_get_high_word( xc, &hx );
-		ihx = (int32_t)hx;
 	}
-	if ( ihx >= HIGH_MAX_NORMAL_EXP ) {
-		return x + x;
+	if ( hx >= HIGH_MAX_NORMAL_EXP ) {
+		return xc + xc;
+	}
+	// Case: log(1) = +0
+	if ( hx == HIGH_BIASED_EXP_0 && lx == 0 ) {
+		return 0;
 	}
-	k += ( ( ihx>>20 ) - STDLIB_CONSTANT_FLOAT64_EXPONENT_BIAS );
-	ihx &= STDLIB_CONSTANT_FLOAT64_HIGH_WORD_SIGNIFICAND_MASK;
-	i = ( ihx + 0x95f64 ) & 0x100000;
+	k += ( ( hx >> 20 ) - STDLIB_CONSTANT_FLOAT64_EXPONENT_BIAS );
+	hx &= STDLIB_CONSTANT_FLOAT64_HIGH_WORD_SIGNIFICAND_MASK;
+	i = ( hx + 0x95f64 ) & HIGH_MIN_NORMAL_EXP;
 
-	// Normalize `x` or `x/2`...
-	stdlib_base_float64_set_high_word( ihx | ( i ^ HIGH_BIASED_EXP_0 ), &xc );
-	k += i >> 20;
+	// Normalize x or x/2...
+	stdlib_base_float64_set_high_word( hx | ( i ^ HIGH_BIASED_EXP_0 ), &xc );
+	k += ( i >> 20 );
 	y = (double)k;
-	f = klog( xc );
-	xc -= 1.0;
-	hi = xc;
+	f = xc - 1.0;
+	hfsq = 0.5 * f * f;
+	R = stdlib_base_kernel_log1p( f );
+
+	/*
+	* Notes:
+	*
+	* -   `f-hfsq` must (for args near `1`) be evaluated in extra precision to avoid a large cancellation when `x` is near `sqrt(2)` or `1/sqrt(2)`.This is fairly efficient since `f-hfsq` only depends on `f`, so can be evaluated in parallel with `R`. Not combining `hfsq` with `R` also keeps `R` small (though not as small as a true `lo` term would be), so that extra precision is not needed for terms involving `R`.
+	* -   When implemented in C, compiler bugs involving extra precision used to break Dekker's theorem for spitting `f-hfsq` as `hi+lo`. These problems are now automatically avoided as a side effect of the optimization of combining the Dekker splitting step with the clear-low-bits step.
+	* -   This implementation uses Dekker's theorem to normalize `y+val_hi`, so, when implemented in C, compiler bugs may reappear in some configurations.
+	* -   The multi-precision calculations for the multiplications are routine.
+	*/
+	hi = f - hfsq;
 	stdlib_base_float64_set_low_word( 0, &hi );
-	lo = xc - hi;
-	z = ( y * LOG10_2LO ) + ( ( xc + f ) * IVLN10LO );
-	z += ( ( lo + f ) * IVLN10HI ) + ( hi * IVLN10HI );
-	return z + ( y * LOG10_2HI );
+	lo = ( f - hi ) - hfsq + R;
+	valHi = hi * IVLN10HI;
+	y2 = y * LOG10_2HI;
+	valLo = ( y * LOG10_2LO ) + ( ( lo + hi ) * IVLN10LO ) + ( lo * IVLN10HI );
+
+	/*
+	* Note:
+	*
+	* -   Extra precision in for adding `y*log10_2hi` is not strictly needed since there is no very large cancellation near `x = sqrt(2)` or `x = 1/sqrt(2)`, but we do it anyway since it costs little on CPUs with some parallelism and it reduces the error for many args.
```

Comment:
While I see that the "in" is present in the [FreeBSD code comments](https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_log10.c?revision=367086&view=markup#l80), it  seems like a typo that should be removed.